src/server/game/Miscellaneous/Formulas.cpp

```diff
@@ -83,7 +83,26 @@ uint32 acore::XP::Gain(Player* player, Unit* unit, bool isBattleGround /*= false
             // xpMod *= creature->GetCreatureTemplate()->ModExperience;
         }
 
-        xpMod *= isBattleGround ? sWorld->getRate(RATE_XP_BG_KILL) : sWorld->getRate(RATE_XP_KILL);
+        if (player->GetSession()->IsPremium())
+        {
+            xpMod *= isBattleGround ? sWorld->getRate(RATE_XP_BG_KILL) : sWorld->getRate(RATE_XP_KILL_PREMIUM);
+        }
+        else
+
+        {
+            xpMod *= isBattleGround ? sWorld->getRate(RATE_XP_BG_KILL) : sWorld->getRate(RATE_XP_KILL);
+        }
```

Comment:
I hope you will [donate to AzerothCore](https://salt.bountysource.com/checkout/amount?team=azerothcore) some of the money you got with this Premium feature...